### PR TITLE
feat: drift apm

### LINKED ISSUE #1566: Support for `drift` APM/crumbs
### Description

https://github.com/simolus3/drift
https://docs.flutter.dev/packages-and-plugins/happy-paths/recommended#background-processing-

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Adds APM integration for [drift](https://pub.dev/packages/drift).

## :bulb: Motivation and Context

Closes #1566 

## :green_heart: How did you test it?
Test suite
Manual testing

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps
Release and docs

### CODE REVIEW

**Comment by buenaflor** [comment]
should be good to go for a review now, the ci package analysis is failing but that's normal for now until we make the first release for the integration

File: dart/lib/src/sentry_trace_origins.dart
Comment (denrase) [inline dart/lib/src/sentry_trace_origins.dart]: Shouldn't the value be named `auto.db.drift.query.executor`? Like it's below for `transaction.executor`?

File: dart/lib/src/sentry_tracer.dart
Comment (denrase) [inline dart/lib/src/sentry_tracer.dart]: Guess this is debug code we can remove?

File: drift/README.md
Comment (denrase) [inline drift/README.md:43]: Should we introduce a helper method to start up the main span around the db? Like we do for `sqflite` with `openDatabaseWithSentry`?

File: drift/README.md
Comment (denrase) [inline drift/README.md:52]: This example won't compile out of the box, right? As the definition for `TodoItem` is not part of it. I'm not sure if this is needed though here and would just bloat the code where we just need to show where the Sentry hook is.

Comment (denrase) [reply drift/README.md:52]: Ignore, it's in a sep. file.

File: drift/example/example.dart
Comment (denrase) [inline drift/example/example.dart:24]: Continuing from above. If it makes sense to provide the main span with a helper, we can also set the origin on it, does that make sense?

File: drift/lib/src/sentry_query_executor.dart
Comment (denrase) [inline drift/lib/src/sentry_query_executor.dart:44]: Does drift use `sqlite` under the hood? Otherwise, we had this discussion in the other PR where we named this value `flutter_hive`, does it make sense here to go with the same naming?

File: drift/lib/src/sentry_span_helper.dart
Comment (denrase) [inline drift/lib/src/sentry_span_helper.dart:74]: This method shares most of the code with `asyncWrapInSpan`. Maybe we can do some reuse/unification.

File: drift/lib/src/sentry_transaction_executor.dart
Comment (denrase) [inline drift/lib/src/sentry_transaction_executor.dart]: No wrap here as well as the remaining methods in this class?

File: drift/lib/src/sentry_transaction_executor.dart
Comment (denrase) [inline drift/lib/src/sentry_transaction_executor.dart]: Calling the constructor is supposed to call `beginTransaction`? Isn't this behaviour we are changing?

**Review by denrase — CHANGES_REQUESTED** [review]
Looking good, mostly questions/suggestions.

- [x] Should we also add sample code to the `flutter/example` app so we & users can easily test the package and check out the created spans on `sentry.io`?

**Comment by buenaflor** [comment]
@denrase 

>  Should we also add sample code to the flutter/example app so we & users can easily test the package and check out the created spans on sentry.io?

Yes, definitely

File: dart/lib/src/sentry_tracer.dart
Comment (buenaflor) [reply dart/lib/src/sentry_tracer.dart]: maybe I'm remembering wrong but don't we have anything in place that would flag stuff like this?

File: drift/lib/src/sentry_query_executor.dart
Comment (buenaflor) [reply drift/lib/src/sentry_query_executor.dart:44]: yup, https://drift.simonbinder.eu/faq/#can-i-view-a-drift-database

since drift itself is not a database but uses sqlite under the hood, I think sqlite should be alright, we also use sqlite for sqflite.

File: drift/lib/src/sentry_span_helper.dart
Comment (buenaflor) [reply drift/lib/src/sentry_span_helper.dart:74]: tbh, since hive also uses the same approach, it might make sense to unify that altogether, what do you think?

File: drift/lib/src/sentry_transaction_executor.dart
Comment (buenaflor) [reply drift/lib/src/sentry_transaction_executor.dart]: Hm, yeah it's not ideal. good catch

File: drift/lib/src/sentry_span_helper.dart
Comment (buenaflor) [reply drift/lib/src/sentry_span_helper.dart:74]: maybe not now but at least as soon as we have merged these

File: dart/lib/src/sentry_trace_origins.dart
Comment (buenaflor) [reply dart/lib/src/sentry_trace_origins.dart]: true

File: drift/README.md
Comment (buenaflor) [reply drift/README.md:43]: afaik sqflite also doesn't work without a transacstion to connect to. `openDatabaseWithSentry` should only be the open span

File: drift/lib/src/sentry_transaction_executor.dart
Comment (buenaflor) [reply drift/lib/src/sentry_transaction_executor.dart]: hm, yes there should be, thx

**Comment by kuhnroyal** [comment]
@buenaflor Are you aware of the drift `QueryInterceptor` which will be added in the next drift version? https://654abec513b45231d6041540--moor.netlify.app/docs/examples/tracing/

**Comment by buenaflor** [comment]
@kuhnroyal I've been talking with Simon (maintainer of drift) about this Sentry integration and he was the one who recommended the approach for this implementation and it looks like he's decided to add an interceptor for cases like these after.

**Comment by buenaflor** [comment]
image

Created spans from the flutter example

**Review by denrase — APPROVED** [review]
LGTM 👍

File: .craft.yml
Comment (denrase) [inline .craft.yml:25]: We probably also need this for the hive PR, right?

Comment (buenaflor) [reply .craft.yml:25]: Yep, we need to do an initial release first and afterwards add it to the registry so the integration can be retrieved in the docs etc.

**Pull request merged**
